### PR TITLE
Adding fcsDataPresent flag

### PR DESCRIPTION
**Pull request by akioogawa**
Adding a flag to tell if FCS data was present or not.

### CODE REVIEW

File: StRoot/StEvent/StFcsCollection.h [hunk 1]
```diff
@@ -64,9 +67,9 @@ class StFcsCollection : public StObject {
     StSPtrVecFcsPoint   mPoints[kFcsNDet];  
 
     Int_t mFcsReconstructionFlag=0;     // undefined for now
+    int mDataExist=0;
```
Comment (klendathu2k) [inline StRoot/StEvent/StFcsCollection.h]: As this is involved in IO, the preference is to use Int_t  instead of int.

**Review by klendathu2k — DISMISSED** [review]
Single change in StFcsCollection requested, otherwise good.

I am not reviewing anything in the pool directory... in fact... I suggest making the changes in the pool area a separate commit so that it is obvious that they have nothing to do with changing our event data model.

File: StRoot/StEvent/StFcsCollection.h (hunk 1 above)
Comment (plexoos) [reply StRoot/StEvent/StFcsCollection.h]: From the name of this variable, it appears it can be boolean

Comment (akioogawa) [reply StRoot/StEvent/StFcsCollection.h]: It's single int per event.... I don't think we need this at the moment, but there are zero-suppressed data and raw-none-zero-suppressed data from DAQ. We only read zero-suppressed data at offline so far, unlike some online reader which reads raw. If we ever needed to read raw, we can turned this int meaning 0=no data, 1=ZS data, 2=raw, 3=embediing...etc. Keeping it Int_t.

**Comment by akioogawa** [comment]
Many test failing with "St_db_Maker::LoadTable(TDataSet*): Assertion `!ee' failed."...  I have no idea. Seems unrelated to my change of 1 line in StEvent...

**Comment by plexoos** [comment]
Those failed jobs are tests with ROOT6. Unrelated to your changes. I restarted them and pretty sure they will pass without issues.

**Review by plexoos — APPROVED** [review]

File: StRoot/StEvent/StFcsCollection.h [hunk 2]
```diff
@@ -64,9 +67,9 @@ class StFcsCollection : public StObject {
     StSPtrVecFcsPoint   mPoints[kFcsNDet];  
 
     Int_t mFcsReconstructionFlag=0;     // undefined for now
```
Comment (zlchang) [inline StRoot/StEvent/StFcsCollection.h:69]: Just a suggestion, maybe put the initialization in the constructor would be better.

File: StRoot/StSpinPool/StFcsRawDaqReader/StFcsRawDaqReader.cxx [hunk 3]
```diff
@@ -154,7 +154,7 @@ Int_t StFcsRawDaqReader::Make() {
       if(trgcmd != 4 && trgcmd !=10){  // 4=phys/ped 10=LED
 	nskip++;
 	nskiptot++;
-	if(mDebug) printf("trgcmd=%d skipping nskip=%d nskiptot=%d\n",trgcmd,nskip,nskiptot);
+	if(mDebug) LOG_INFO << Form("trgcmd=%d skipping nskip=%d nskiptot=%d",trgcmd,nskip,nskiptot)<<endm;
```
Comment (zlchang) [inline StRoot/StSpinPool/StFcsRawDaqReader/StFcsRawDaqReader.cxx:157]: Is there a need to introduce mDebug, maybe just use LOG_DEBUG<<Form(...)<<endm;?

File: StRoot/StSpinPool/StFcsRawDaqReader/StFcsRawDaqReader.cxx [hunk 4]
```diff
@@ -330,7 +333,7 @@ Int_t StFcsRawDaqReader::Make() {
       if(detid<6) nvaliddata++;      
       
       if(mDebug){
-	printf("FCS %3s : S%d:%d [det %d, ns %d, dep %d ch %d] det=%d id=%3d : size=%d : adc=",
+	printf("FCS %3s : S%2d:%2d [det%1d ns%1d dep%2d ch%2d] det=%d id=%3d size=%3d adc=",
```
Comment (zlchang) [inline StRoot/StSpinPool/StFcsRawDaqReader/StFcsRawDaqReader.cxx:336]: Same as here, LOG_DEBUG?

File: StRoot/StSpinPool/StFcsRawDaqReader/StFcsRawDaqReader.cxx [hunk 5]
```diff
@@ -353,7 +356,7 @@ Int_t StFcsRawDaqReader::Make() {
     dd = mRdr->det("stgc")->get("altro",r) ;
     while(dd && dd->iterate()) {    //per xing and per RDO    
       if(mDebug) printf("STGC ALTRO: stgc%02d(sec) RDO=%1d ALTRO=%03d(row) Ch=%02d(pad)\n",dd->sec,r,dd->row,dd->pad);
-      for(uint32_t i=0; i<dd->ncontent; i++) {
+      for(u_int i=0; i<dd->ncontent; i++) {
 	if(mDebug) printf(" TB=%3d ADC=%4d",dd->adc[i].tb,dd->adc[i].adc) ;
```
Comment (zlchang) [inline StRoot/StSpinPool/StFcsRawDaqReader/StFcsRawDaqReader.cxx:360]: LOG_DEBUG and the same applies to a few lines below if(mDebug) printf("\n")?

**Review by zlchang — APPROVED** [review]

**Review by klendathu2k — APPROVED** [review]
StEvent changes are fine.  Generally I agree w/ Zilong's comments about LOG_DEBUG the mDebug parameter.  But not a show stopper ... especially for a "pool" code.

**Pull request merged**